core/src/main/scala/org/broadinstitute/dsde/rawls/user/UserService.scala

```diff
@@ -205,38 +222,39 @@ class UserService(protected val ctx: RawlsRequestContext,
       }
     } yield constructBillingProjectResponseFromOptionalAndRoles(maybeBillingProject, projectRoles)
 
-  private def makeBillingProjectResponse(projectRoles: Set[ProjectRole], billingProject: RawlsBillingProject) =
-    RawlsBillingProjectResponse(
-      billingProject.projectName,
-      billingProject.billingAccount,
-      billingProject.servicePerimeter,
-      billingProject.invalidBillingAccount,
-      projectRoles,
-      billingProject.status,
-      billingProject.message,
-      billingProject.azureManagedAppCoordinates
-    )
-
-  def listBillingProjectsV2(): Future[List[RawlsBillingProjectResponse]] =
-    for {
-      samUserResources <- samDAO.listUserResources(SamResourceTypeNames.billingProject, ctx.userInfo)
-      projectNames = samUserResources.map(r => RawlsBillingProjectName(r.resourceId)).toSet
-      projectsInDB <- dataSource.inTransaction { dataAccess =>
-        dataAccess.rawlsBillingProjectQuery.getBillingProjects(projectNames)
-      }
-      bpmProfiles <- billingProfileManagerDAO.listBillingProfiles(samUserResources, ctx)
-    } yield constructBillingProjectResponses(samUserResources, projectsInDB ++ bpmProfiles)
-
-  private def constructBillingProjectResponses(samUserResources: Seq[SamUserResource],
-                                               billingProjectsInRawlsDB: Seq[RawlsBillingProject]
-  ): List[RawlsBillingProjectResponse] = {
-    val projectsByName = billingProjectsInRawlsDB.map(p => p.projectName.value -> p).toMap
-    samUserResources.toList
-      .flatMap { samUserResource =>
-        val projectRoles = samRolesToProjectRoles(samUserResource.direct.roles ++ samUserResource.inherited.roles)
-        projectsByName.get(samUserResource.resourceId).map(makeBillingProjectResponse(projectRoles, _))
-      }
-      .sortBy(_.projectName.value)
+  def listBillingProjectsV2(): Future[List[RawlsBillingProjectResponse]] = for {
+    samUserResources <- samDAO.listUserResources(SamResourceTypeNames.billingProject, ctx.userInfo)
```

Comment:
I think this may be a good candidate for extraction to the `BillingProjectOrchestrator` class and unit level coverage, given the amount of business logic in play here. Thoughts?

Reply:
I do think the billing _package_ is a good place for this.

Not sure about `BillingProjectOrchestrator` though. It seems to be more about managing multiple-step creation, and being able to roll things back, etc. I feel like this doesn't really fit, and risks bloating that class.

I'll poke around and see if there's somewhere where it fits.

Reply:
I see why you were thinking about putting it there, now.... there isn't really a different billing specific service...

Reply:
I agree that `BillingProjectOrchestrator` didn't feel like the right place to me, or if it is, the documentation/concept of it needs to change. But I do think we should put it someplace where we can get some test coverage… the code is certainly non-trivial.

I would advocate for doing this work in a new ticket so we can close this one out this sprint and continue making progress on putting the frontend and backend together.